### PR TITLE
only extract 1000 frames at a time to prevent ffmpeg failing

### PR DESCRIPTION
**Pull request by pfeiferj**
Currently whenever the script tries to extract a very large number of frames (greater than approximately 3000) it will fail and report that ffmpeg is missing. This PR applies a simple workaround of calling ffmpeg once for each group of 1000 images.

### CODE REVIEW

**Comment by pfeiferj** [comment]
For some additional context,
I uploaded some 360 videos with the desktop uploader and they failed to process. I decided to try out the command-line tools to locally process the videos and noticed that on some videos it was saying ffmpeg was not found while others were fine. As i was running on windows I figured it was just a command text length issue but then saw that there was already a workaround for that.  At that point I figured it was probably an ffmpeg limitation and went ahead with creating this fix. Tested the fix on a file that failed previously and it successfully uploaded approximately 3500 images.

**Comment by pfeiferj** [comment]
Here's a sequence that failed in case you also want to check the server side: 591066202938634

File: mapillary_tools/ffmpeg.py [hunk 1]
```diff
@@ -231,54 +232,57 @@ def extract_specified_frames(
...
-                self._run_ffmpeg(cmd)
-            finally:
-                if not delete:
-                    try:
-                        os.remove(select_file.name)
-                    except FileNotFoundError:
-                        pass
+        eqs = [f"eq(n\\,{idx})" for idx in sorted(frame_indices)]
+
+        for i in range(0, ceil(len(eqs) / 1000)):
```
Comment (ptpt) [inline mapillary_tools/ffmpeg.py:237]: There is an util function `group_every` in geo.py that does exactly the same thing. Could you try:

```python
for partial_indices in group_every(sorted(frame_indices), 3000):
    eqs = [f"eq(n\\,{idx})" for idx in sorted(partial_indices)]
```

Comment (ptpt) [reply mapillary_tools/ffmpeg.py:237]: Also I'd suggest try make it the group size maximum as possible. I had the impression that ffmpeg sampling is a lot slower when extracting later frames in a stream, so we should extract as many frames as possible in one run.

**Review by ptpt — CHANGES_REQUESTED** [review]
use group_every

**Comment by ptpt** [comment]
@pfeiferj Is it still an issue for Windows after this fix https://github.com/mapillary/mapillary_tools/pull/601?

**Comment by pfeiferj** [comment]
> @pfeiferj Is it still an issue for Windows after this fix https://github.com/mapillary/mapillary_tools/pull/601?

Ah, my bad, I had thought that fix was already in the latest release. Retested and made sure I was on main branch and didn't have the issue. Going to close the PR

**Pull request closed without merging**